medcat/cat.py

```diff
@@ -458,24 +459,55 @@ def _print_stats(self,
 
         return fps, fns, tps, cui_prec, cui_rec, cui_f1, cui_counts, examples
 
-    def train(self, data_iterator: Iterable, fine_tune: bool = True, progress_print: int = 1000) -> None:
+    def train(self,
+              data_iterator: Iterable,
+              fine_tune: bool = True,
+              progress_print: int = 1000,
+              ckpt_dir_path: str = os.path.join(os.path.abspath(os.getcwd()), "checkpoints", "cat_train"),
+              ckpt_steps: int = 1000,
+              ckpt_max_to_keep: int = 1,
+              num_of_skipped: int = 0,
+              is_resumed: bool = False) -> None:
         """ Runs training on the data, note that the maximum length of a line
         or document is 1M characters. Anything longer will be trimmed.
 
-        data_iterator:
-            Simple iterator over sentences/documents, e.g. a open file
-            or an array or anything that we can use in a for loop.
-        fine_tune:
-            If False old training will be removed
-        progress_print:
-            Print progress after N lines
+        Args:
+            data_iterator (Iterable):
+                Simple iterator over sentences/documents, e.g. a open file
+                or an array or anything that we can use in a for loop.
+            fine_tune (bool):
+                If False old training will be removed.
+            progress_print (int):
+                Print progress after N lines.
+            ckpt_dir_path (str):
+                The path to the checkpoint directory.
+            ckpt_steps (int):
+                The number of processed sentences/documents before checkpoint is created.
+                N.B.: A smaller number could result in the out-of-disk-space error.
+            ckpt_max_to_keep (int):
+                The maximum number of checkpoints to keep.
+                N.B.: A larger number could result in the out-of-disk-space error.
+            num_of_skipped (int):
+                The number of sentences/documents to ignore at the beginning.
+            is_resumed (bool):
+                Used for resuming a previous training (True) or starting a fresh new training (False).
         """
+        if ckpt_max_to_keep < 1:
```

Comment:
Do not like this, I've rarely (or never) seen this feature

Reply:
Checks like this are now wrapped in the decorator.